fontforge/cvexport.c

```diff
@@ -702,19 +702,29 @@ static void MakeExportName(char *buffer, int blen,char *format_spec,
 	    ++format_spec;
 	    ch = *format_spec++;
 	    if ((bend = buffer+40)>end ) bend = end;
-	    if ( ch=='n' ) {
+	    if ( ch=='n' || ch=='N' ) {
+            pt=sc->name;
+            if (ch=='N') {
+                ligature = GetLigature(sc);
+                if (ligature != NULL) {
+                    pt = ligature;
+                }
+            }
 #if defined( __CygWin ) || defined(__Mac)
 		/* Windows file systems are not case conscious */
 		/*  nor is the default mac filesystem */
-		for ( pt=sc->name; *pt!='\0' && buffer<bend; ) {
+		for ( ; *pt!='\0' && buffer<bend; ) {
 		    if ( isupper( *pt ))
 			*buffer++ = '$';
 		    *buffer++ = *pt++;
 		}
 #else
-		for ( pt=sc->name; *pt!='\0' && buffer<bend; )
+		for ( ; *pt!='\0' && buffer<bend; )
 		    *buffer++ = *pt++;
 #endif
+        if (ligature != NULL) {
```

Comment:
You just did what I said wasn't safe in my comment. It's not safe for two reasons:

1) If ch is not "N" before this line is reached (say it's "n"), then the ligature pointer will never be initialized. This could be fixed by initializing ligature to "NULL", but ... 
2) If ch has been "N" in the past and there was a ligature name, but it isn't currently "N", the `ligature` pointer will point to the previous entry, and this code will double-free it. 

Also note that freeing NULL is fine, the function knows to do nothing, so you never have to check for NULL prior to calling it. 

If you want to do this style of freeing, you would 

1. Set ligature to NULL at the top, and
2. Set ligature to NULL right after freeing it. 

That way it will only be non-NULL from a return from `GetLigature()`, and only for the lifetime of a single iteration.